dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -106,13 +134,25 @@ public static void registerZenodoDOIForWorkflow(String zenodoUrl, String docksto
                 // accessed through the "latest_draft" under "links" in the response body.
                 Object links = returnDeposit.getLinks();
                 String depositURL = (String)((LinkedHashMap)links).get("latest_draft");
-
                 String depositionIDStr = depositURL.substring(depositURL.lastIndexOf("/") + 1).trim();
                 // Get the deposit object for the new workflow version DOI
                 depositionID = Integer.parseInt(depositionIDStr);
                 returnDeposit = depositApi.getDeposit(depositionID);
 
                 depositMetadata = returnDeposit.getMetadata();
+
+                // Retrieve the DOI so we can use it to create a Dockstore alias
+                // to the workflow; we will add that alias as a Zenodo related identifier
+                String doi = depositMetadata.getDoi();
+                String doiAlias = createAliasUsingDoi(doi, workflow);
+
+                // Add the new alias created with the DOI to the deposit metadata
+                // Later on we will put the deposit on Zenodo in the call to
+                // putDepositionOnZenodo so it contains the workflow version
+                // alias constructed with the DOI
+                setMetadataRelatedIdentifiers(depositMetadata, dockstoreBaseUrl, dockstoreUrl,
+                        workflowUrl, workflow, workflowVersion, doiAlias);
+
                 fillInMetadata(depositMetadata, dockstoreUrl, workflow, workflowVersion);
```

Comment:
Just making sure -- here you call `fillInMetadata` after calling `setMetadataRelatedIdentifiers`; in the block above you call `fillInMetadata` before calling `setMetadataRelatedIdentifiers`. Is that intentional and necessary?

Reply:
New method createAliasAndsetUpRelatedIdentifiers fixes this

Reply:
I still see `fillInMetadata` called in different order in the two blocks. Is this intentional?

Reply:
I still see it called in different order. Is this intentional?